### DISCUSSION TITLE
Classification Menggunakan Decision Tree & Random Forest

### DISCUSSION THREAD

**Discussion by joejnt** [teamalgoritma/community#537] [Q&A] [answered]
Dear rekan2 algoritma,

Mohon saran dan petunjuknya, 
Saat melakukan proses EDA, saya mendapatkan dataset dengan korelasi yang mayoritas positif kuat dengan plot seperti ini:

![image](https://user-images.githubusercontent.com/103427989/217465198-bfe4abd3-0888-4ae1-86d6-bff88995a943.png)

Saya membaca artikel mendapatkan "Koefisien korelasi yang tinggi (mendekati 1 untuk hubungan linier yang searah atau mendekati -1 untuk hubungan linier yang berlawanan arah) merupakan indikasi redundancy".

Terkait dengan kondisi tersebut, treatment nya seperti apa ya, apakah variabel yang termasuk redundancy perlu dibuang atau dipertahankan, dengan kondisi observasi pada dataset hanya 110 dengan jumlah variable prediktor 37.

Mohon saran dan pendapatnya.

Best regards,
JT

**Comment by muhrisman** [comment, marked as answer]
Model yang baik adalah model yang mampu mendapatkan akurasi yang tinggi dengan variabel yang secukupnya.
Dalam kasus itu banyak variabel yang memiliki korelasi yang tinggi satu sama lain, saya memprediksi banyak variabel yang merupakan kombinasi matematis dari variabel lain.
Ada baiknya coba dikurangi bapak untuk variabel - variabel tersebut untuk mengurangi redudansi informasi.

**Reply by joejnt** [reply]
Dear @muhrisman 

Noted, terima kasih untuk insidenya, untuk menguranngi redudansi, apakah bisa ditreatment dengan menggunakan PCA ?

Best regards,
JT

**Reply by kevbow** [reply]
Dear @joejnt,

Pada konsepnya memang PCA digunakan untuk data-data yang memiliki korelasi tinggi sehingga pada kasus Anda hal ini bisa digunakan. Korelasi tinggi dapat dikatakan memiliki informasi redundant.

**Comment by joejnt** [comment]
Dear @kevbow 

Noted, terima kasih untuk insidenya, need advice apakah punya referensi untuk membuat pemodelan credit scoring ?

really need ur advice and recomendation.

Best regards,
JT

**Reply by finesaaa** [reply]
Dear Pak @joejnt,
Kami ada beberapa referensi terkait pembuatan model credit scoring menggunakan R
- Disini ada video kami "CREDIT SCORING | Algoritma Show | Episode #1 | Algoritma 2021", https://youtu.be/L7564DMRcdY terkait pembuatan model credit scoring menggunakan R
Selain itu ada beberapa referensi lainnya
- https://rpubs.com/ngyongkad/scorecard
- https://cran.r-project.org/doc/contrib/Sharma-CreditScoring.pdf
- https://www.analyticsvidhya.com/blog/2019/03/introduction-creditr-r-package-enhance-credit-risk-scoring-validation-r-codes/
Semoga membantu Pak, sekian.

**Reply by joejnt** [reply]
Dear @finesaaa 

Terima kasih untuk referensinya.

Best regards,
JT
